tianshou/data/buffer.py

```diff
@@ -520,13 +437,215 @@ def update_weight(
     def __getitem__(
         self, index: Union[slice, int, np.integer, np.ndarray]
     ) -> Batch:
-        return Batch(
-            obs=self.get(index, "obs"),
-            act=self.act[index],
-            rew=self.rew[index],
-            done=self.done[index],
-            obs_next=self.get(index, "obs_next"),
-            info=self.get(index, "info"),
-            policy=self.get(index, "policy"),
-            weight=self.weight[index],
-        )
+        batch = super().__getitem__(index)
+        batch.weight = self.weight[index]
+        return batch
+
+
+class ReplayBuffers(ReplayBuffer):
+    """ReplayBuffers contains a list of ReplayBuffer.
+
+    These replay buffers have contiguous memory layout, and the storage space
+    each buffer has is a shallow copy of the topmost memory.
+
+    :param int buffer_list: a list of ReplayBuffers needed to be handled.
+
+    .. seealso::
+
+        Please refer to :class:`~tianshou.data.ReplayBuffer` for more detailed
+        explanation.
+    """
+
+    def __init__(self, buffer_list: List[ReplayBuffer], **kwargs: Any) -> None:
+        self.buffer_num = len(buffer_list)
+        self.buffers = buffer_list
+        offset = 0
+        for buf in self.buffers:
+            buf.alloc_fn = self.alloc_fn  # type: ignore
+            buf.offset = offset
+            offset += buf.maxsize
+        super().__init__(size=offset, **kwargs)
+
+    def __len__(self) -> int:
+        return sum([len(buf) for buf in self.buffers])
+
+    def reset(self) -> None:
+        for buf in self.buffers:
+            buf.reset()
```

Comment:
should we delete self._index, _size, _episode_reward, _episode_length, _save_only_last_obs, _save_obs_next, _indices since they are never used?

Reply:
No, self._save_obs_next and self._indices is needed because ReplayBuffers inherits the bsae class's `__getitem__`. If it is removed, the cached replay buffer cannot work as we expected.

Reply:
what about self._index, _size,_episode_reward, _episode_length, then